src/cdomains/intDomain.ml

```diff
@@ -911,30 +938,600 @@ struct
+    let is_false = fun x -> x = zero in
+    let is_true =  fun x -> x = one in
+    if is_true x then zero else if is_false x then one else top_bool 
+
+  let to_bool = function  
+    | [(l,u)] when Ints_t.compare l Ints_t.zero = 0 && Ints_t.compare u Ints_t.zero = 0 -> Some false
+    | x -> if leq zero x then None else Some true
+
+  let of_bool _ = function true -> one | false -> zero 
+
+  let of_interval ?(suppress_ovwarn=false) ik (x,y) =  norm_interval  ~suppress_ovwarn ~cast:false ik (x,y)
+
+  let of_int ik (x: int_t) = of_interval ik (x, x)
+
+  let lt ik x y = 
+    match x, y with 
+    | [], [] -> bot_of ik
+    | [], _ | _, [] -> raise (ArithmeticOnIntegerBot (Printf.sprintf "%s op %s" (show x) (show y)))
+    | _, _ ->
+      let (max_x, min_y) = (maximal x |> Option.get , minimal y |> Option.get) in
+      let (min_x, max_y) = (minimal x |> Option.get , maximal y |> Option.get) in
+      if max_x < min_y then 
+        of_bool ik true
+      else
+        if min_x >= max_y then of_bool ik false else top_bool
+
+  let le ik x y =
+    match x, y with 
+    | [], [] -> bot_of ik
+    | [], _ | _, [] -> raise (ArithmeticOnIntegerBot (Printf.sprintf "%s op %s" (show x) (show y)))
+    | _, _ ->
+      let (max_x, min_y) = (maximal x |> Option.get , minimal y |> Option.get) in
+      let (min_x, max_y) = (minimal x |> Option.get , maximal y |> Option.get) in
+      if max_x <= min_y then 
+        of_bool ik true
+      else
+        if min_x > max_y then of_bool ik false else top_bool
+
+  let gt ik x y = not_bool @@ le ik x y
+
+  let ge ik x y = not_bool @@ lt ik x y
+
+  let eq ik x y = match x, y with
+    | (a, b)::[], (c, d)::[] when (Ints_t.compare a b) == 0 && (Ints_t.compare c d) == 0 && (Ints_t.compare a c) == 0 -> 
+      one
+    | _ -> 
+      if is_bot (meet ik x y) then 
+        zero 
+      else 
+        top_bool
+
+  let ne ik x y = not_bool @@ eq ik x y
+  let interval_to_int i = Interval.to_int (Some i)
+  let interval_to_bool i = Interval.to_bool (Some i)
+
+  let log f ik (i1, i2) = 
+    match (interval_to_bool i1, interval_to_bool i2) with
+    | Some x, Some y -> of_bool ik (f x y)
+    | _ -> top_of ik
+
+  let log1 f ik i1 = 
+    match interval_to_bool i1 with
+    | Some x -> of_bool ik (f ik x)
+    | _ -> top_of ik
+
+  let bit f ik (i1, i2) =
+    match (interval_to_int i1), (interval_to_int i2) with
+    | Some x, Some y -> (try of_int ik (f ik x y) |> unlift with Division_by_zero -> top_of ik)
+    | _ -> top_of ik
+
+  let bit1 f ik i1 =
+    match interval_to_int i1 with
+    | Some x -> of_int ik (f ik x) |> unlift
+    | _ -> top_of ik
+
+  let bitcomp f ik (i1, i2) = 
+    match (interval_to_int i1, interval_to_int i2) with 
+    | Some x, Some y -> (try  of_int ik (f ik x y) with Division_by_zero | Invalid_argument _ -> (top_of ik,false,false,false))
+    | _, _ -> (top_of ik,true,true,false)
+
+  let bitand ik x y = 
+    let interval_bitand = bit (fun _ik -> Ints_t.bitand) ik in
```

Comment:
You may redefine `bit` such that the `f` that it takes does not take an `ikind`.